plugins/eni/types/types_test.go

```diff
@@ -22,116 +22,86 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-var validConfigNoIPV6 = `{"eni":"eni1", "ipv4-address":"10.11.12.13", "mac":"01:23:45:67:89:ab"}`
-var configNoENIID = `{"ipv4-address":"10.11.12.13", "mac":"01:23:45:67:89:ab"}`
-var configNoIPV4Address = `{"eni":"eni1", "mac":"01:23:45:67:89:ab"}`
-var configNoMAC = `{"eni":"eni1", "ipv4-address":"10.11.12.13"}`
-var configInvalidIPV4AddressMalformed = `{"eni":"eni1", "ipv4-address":"1", "mac":"01:23:45:67:89:ab"}`
-var configInvalidIPV4AddressIPV6 = `{"eni":"eni1", "ipv4-address":"2001:db8::68", "mac":"01:23:45:67:89:ab"}`
-var configMalformedMAC = `{"eni":"eni1", "ipv4-address":"10.11.12.13", "mac":"01:23:45:67:89"}`
-var validConfigWithIPV6 = `{
+var (
+	validConfigNoIPV6                 = `{"eni":"eni1", "ipv4-address":"10.11.12.13", "mac":"01:23:45:67:89:ab"}`
+	configNoENIID                     = `{"ipv4-address":"10.11.12.13", "mac":"01:23:45:67:89:ab"}`
+	configNoIPV4Address               = `{"eni":"eni1", "mac":"01:23:45:67:89:ab"}`
```

Comment:
Is it still an invalid config if no IPv4 address is specified but stay-down is set to true? It feels like IPAddress fields should be optional when stay-down==true. I guess agent will never leave the IPv4 address field empty, so this is a pureness argument rather than an actual requirement, so I'm fine with the way the code is now.

Reply:
> Is it still an invalid config if no IPv4 address is specified but stay-down is set to true?

I think that config should be valid. But, I don't know if we need to relax these constraints yet. As you point out, these fields are being populated by the agent anyway. We can cross that bridge when we need to I guess.